pallets/proofs-dealer/src/lib.rs

```diff
@@ -388,3 +387,10 @@ impl InherentError {
 pub trait TrieVerifier {
     fn verify_proof(root: &[u8; 32], challenges: &[u8; 32], proof: &CompactProof) -> bool;
 }
+
+impl<T: Config> TrieVerifier for Pallet<T> {
```

Comment:
While I don’t see the exact reason for the pallet not being able to implement this, I also don’t see any reason to not implement it can outside of it as well.